notebooks/oracle2bq/OracleToBigQuery_notebook.ipynb

```diff
@@ -0,0 +1,888 @@
+    "            ORACLETABLE_LIST.append(row[0])\n",
+    "\n",
+    "print(\"list of tables for migration :\")\n",
+    "print(ORACLETABLE_LIST)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1d9a62e8-7499-41c6-b32b-73b539b0c7c4",
+   "metadata": {},
+   "source": [
+    "## Step 5: Get Primary Keys for tables from ORACLE source\n",
+    "This step fetches primary key of tables listed in ORACLETABLE_LIST from ORACLE_DATABASE"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e7b9074c-a38e-4261-ae4e-57a498aaec88",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "SQL_TABLE_PRIMARY_KEYS = {} #dict for storing primary keys for each table"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "6eda8fac-582c-4d4a-b871-311bb2863335",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "DB = sqlalchemy.create_engine(\n",
+    "            sqlalchemy.engine.url.URL.create(\n",
+    "                drivername=ORACLE_DRIVER,\n",
+    "                username=ORACLE_USERNAME,\n",
+    "                password=ORACLE_PASSWORD,\n",
+    "                database=ORACLE_DATABASE,\n",
+    "                host=ORACLE_HOST,\n",
+    "                port=ORACLE_PORT\n",
+    "              )\n",
+    "            )\n",
+    "with DB.connect() as conn:\n",
+    "    for table in ORACLETABLE_LIST:\n",
+    "        primary_keys = []   \n",
+    "        results = DB.execute(\"SELECT cols.column_name  FROM sys.all_constraints cons, sys.all_cons_columns cols WHERE cols.table_name = '{}' AND cons.constraint_type = 'P' AND cons.status = 'ENABLED' AND cols.position = 1 AND cons.constraint_name = cols.constraint_name AND cons.owner = cols.owner\".format(table)).fetchall()\n",
+    "        for row in results:\n",
+    "            primary_keys.append(row[0])\n",
+    "        if primary_keys:\n",
+    "            SQL_TABLE_PRIMARY_KEYS[table] = \",\".join(primary_keys)\n",
+    "        else:\n",
+    "            SQL_TABLE_PRIMARY_KEYS[table] = \"\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7c2a210f-48da-474f-bf46-89e755d01c67",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "pkDF = pd.DataFrame({\"table\" : ORACLETABLE_LIST, \"primary_keys\": list(SQL_TABLE_PRIMARY_KEYS.values())})\n",
+    "print(\"Below are identified primary keys for migrating ORACLE table to Bigquery:\")\n",
+    "pkDF"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f8ac9a9f-1332-4c75-b6ca-bfb3b776671f",
+   "metadata": {},
+   "source": [
+    "# Step 6: Get Row Count of Tables and identify Partition Columns\n",
+    "This step uses PARTITION_THRESHOLD(default value is 1 million) parameter and any table having rows greater than PARTITION_THRESHOLD will be used for partitioned read based on Primary Keys\n",
+    " - NUMBER_OF_PARTITIONS : Number of partitioned reads default 10\n",
```

Comment:
NUMBER_OF_PARTITIONS should not be hardcoded.
Refer to MySQLToSpanner notebook for latest code.